galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -353,3 +358,94 @@ def _remove_content(collection_version, repo):
+            self.get_object(),
+            context=self.get_serializer_context()
+        )
+
+        dependency_map = {}
+        resolved_deps = {}
+
+        # Initial dependencies
+        deps = serializer.data['dependencies']
+
+        while len(deps) > 0:
+            resolved_version_number = ''
+            new_dependencies = {}
+            current_dep = ''
+
+            for key in deps:
+                current_dep = key
+
+                # Add dependencies to dependency map
+                if key in resolved_deps and key in dependency_map:
+                    resolved_deps.pop(key)
+                    dependency_map[key].append(deps[key])
+                    deps[key] += ',' + ','.join(dependency_map[key])
+                else:
+                    if key in dependency_map:
+                        dependency_map[key].append(deps[key])
+                        deps[key] = ','.join(dependency_map[key])
+                    else:
+                        dependency_map[key] = [deps[key]]
+
+                # Build then run sql query based on deps to get the latest compatible version
+                namespace, name = key.split('.')
+                split_deps = deps[key].split(',')
+                filters = []
+                for d in split_deps:
+                    offset = 0
+                    if '>=' in d or '<=' in d:
+                        offset = 2
+                    if '=' in d or '>' in d or '<' in d:
+                        offset = 1
+                    operand = d[:offset]
+                    version = d[offset:]
+                    if '>=' == operand:
+                        filters.append(Q(version__gte=version))
+                    elif '<=' == operand:
+                        filters.append(Q(version__lte=version))
+                    elif '=' == operand:
+                        filters.append(Q(version__eq=version))
+                    elif '>' == operand:
+                        filters.append(Q(version__gt=version))
+                    elif '<' == operand:
+                        filters.append(Q(version__lt=version))
+                    else:
+                        filters.append(Q(version__gt=0))
+
+                qs = CollectionVersion.objects.filter(
+                    reduce(AND, filters),
+                    namespace__in=[namespace],
+                    name__in=[name]
+                ).order_by('-version')[:1]
+
+                if qs:
+                    results = self.get_serializer_class()(qs[0])
+                    resolved_version_number = results.data['version']
+                    new_dependencies = results.data['dependencies']
+                else:
+                    resolved_version_number = False
+                    new_dependencies = []
+
+            if resolved_version_number:
+                resolved_deps[key] = resolved_version_number
+            else:
+                resolved_deps[key] = f"Conflict! {deps}"
```

Comment:
In case of dependency conflict it does not produce an error, but returns formatted string. Could you please give examples of API response format?

Reply:
With conflict:
```
{
    "bmclaughlin.conflict2": "0.0.3",
    "bmclaughlin.deps2": "Conflict! {'bmclaughlin.deps2': '>0.1.0,<0.1.0'}"
}
```
Without conflict:
```
{
    "bmclaughlin.deps2": "0.0.3",
    "bmclaughlin.deps1": "0.0.4",
    "bmclaughlin.execution_env_test": "0.1.2",
    "ansible.netcommon": "1.3.0"
}
```

Reply:
Response format in case of conflict is vague and requires parsing on a client.

Can you please give a link to this API format specification?